### PR TITLE
move session management from get_or_create_user to authenticate

### PR DESCRIPTION
**Pull request by anlutro**
This is something which was introduced in https://github.com/mozilla/mozilla-django-oidc/pull/179.

I think it's wrong to do anything with the session in a `get_or_create_user` method.

Hopefully this shouldn't affect anyone overriding the method(s).

### CODE REVIEW

**Comment by anlutro** [comment]
Tests seem to fail on Travis because `wheel` stopped supporting Python 3.3?

**Comment by johngian** [comment]
@anlutro Tests should be fixed after #239 can you rebase ?

**Comment by anlutro** [comment]
@johngian done!

**Comment by johngian** [comment]
Please change the commit name to something more specific since we are not actually handling the session management in general but specifically storing the OIDC tokens.

**Comment by johngian** [comment]
r+wc

File: mozilla_django_oidc/auth.py [hunk 1]
```diff
@@ -247,15 +255,6 @@ def get_or_create_user(self, access_token, id_token, verified_id):
         """Returns a User instance if 1 user is found. Creates a user if not found
         and configured to do so. Returns nothing if multiple users are matched."""
 
-        session = self.request.session
-
-        if import_from_settings('OIDC_STORE_ACCESS_TOKEN', False):
```
Comment (johngian) [inline mozilla_django_oidc/auth.py]: Ideally i would prefer we move storing tokens to another method since it's quite self contained.

**Comment by anlutro** [comment]
Amended to "write oidc tokens to session in authenticate(), not get_or_create_user()"

File: mozilla_django_oidc/auth.py (hunk 1 above)
Comment (anlutro) [reply mozilla_django_oidc/auth.py]: Sounds reasonable, will do that.

**Comment by anlutro** [comment]
Extracted a `_write_tokens_to_session` method.

**Comment by johngian** [comment]
Closing this one after #241.

Thanks @anlutro for the contribution :rocket:

**Pull request closed without merging**